### PR TITLE
Add isometric camera

### PR DESCRIPTION
**Pull request by shawwwwwwww**

### CODE REVIEW

**Comment by shawwwwwwww** [comment]
@hamzzam can you pull dev in and make sure this is up to date if it's finished?

File: packages/engine/src/camera/systems/CameraSystem.ts [hunk 1]
```diff
@@ -51,6 +51,17 @@ const followCameraBehavior = (entity: Entity) => {
     FollowCameraComponent
   ) as FollowCameraComponent
 
+  if (CameraSystem.instance.updateCameraMode) {
+    let index = 0
+    for (const key of Object.keys(CameraModes)) {
+      if (index === CameraSystem.instance.cameraModeIndex && CameraSystem.instance.updateCameraMode) {
```
Comment (NPatel10) [inline packages/engine/src/camera/systems/CameraSystem.ts:57]: Why are we checking `CameraSystem.instance.updateCameraMode` here? We have already checked on the if condition above.

**Comment by HexaField** [comment]
We should really change how we are handling the camera modes in the engine. I think instead of one follow camera component, we should have a separate component for each of the modes, and then also the FollowCameraComponent. This way, input behaviors just add and remove components, and all camera logic can be handled in the camera system. This also allows more modularity with how the camera system operates.

**Comment by shawwwwwwww** [comment]
> We should really change how we are handling the camera modes in the engine. I think instead of one follow camera component, we should have a separate component for each of the modes, and then also the FollowCameraComponent. This way, input behaviors just add and remove components, and all camera logic can be handled in the camera system. This also allows more modularity with how the camera system operates.

I agreed, but we need to get this in for Mappa

**Pull request merged**